### PR TITLE
Add support for Realsense Depth Camera for Obstacle Avoidance

### PR DESCRIPTION
**Pull request by thien94**
The main content is extracted and simplified from the following blog posts: [1](https://discuss.ardupilot.org/t/gsoc-2020-enhancements-of-non-gps-navigation-and-object-avoidance-with-realsense-cameras/55984), [2](https://discuss.ardupilot.org/t/gsoc-2020-integration-of-ardupilot-and-realsense-d4xx-depth-camera-for-simple-obstacle-avoidance/58474/1), [3](https://discuss.ardupilot.org/t/gsoc-2020-enhancements-of-gps-less-navigation-and-obstacle-avoidance-with-tracking-and-depth-camera/60285).

Additional changes:
- Replacement of the `sphinxcontrib.vimeo` (the original link cannot be found anymore) with @khancyr 's https://github.com/khancyr/sphinxcontrib.vimeo.git
- Update the git links of the T265 wiki page.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
This builds without error locally...however several questions need to be resolved before merge:
- [ ] Randy, please review for content accuracy
- [ ] Pierre, please confirm review of this and although the vimeo display plugin should be in a separate commit, my main concern is that it would seem to leave a repo laying around...perhaps something like git+https....? Also, for local wiki builders, after this is merged, should they re-reun Sphinxsetup.sh?

**Comment by khancyr** [comment]
I have already open a PR with the full change need to support the new target ok vimeo plugin. As it seems to be needed we will create a repo into ArduPilot GitHub and not in mine.
It won't impact people that already have the plug-in installed

**Comment by Hwurzburg** [comment]
thanks....okay, so when time comes, and I squash before merge, I will drop the Sphinx commit from this PR....

File: common/source/docs/common-realsense-depth-camera.rst [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    # Only necessary if you installed the minimal version of Ubuntu
+    sudo apt install python3-opencv
+
+
+Download the main script `d4xx_to_mavlink.py <https://github.com/thien94/vision_to_mavros/blob/master/scripts/d4xx_to_mavlink.py>`__ or clone the `vision_to_mavros <https://github.com/thien94/vision_to_mavros>`__ repository and find the script folder.
+
+.. code-block:: bash
+
+    # Clone the package in the ROS workspace or any location if you don't use ROS
+    cd ~/catkin_ws/src
```
Comment (rmackay9) [inline common/source/docs/common-realsense-depth-camera.rst:116]: not using ROS so do we really want to put these scripts in the catkin directory?

**Comment by rmackay9** [comment]
This looks good to me, thanks!

File: common/source/docs/common-realsense-depth-camera.rst (hunk 1 above)
Comment (thien94) [reply common/source/docs/common-realsense-depth-camera.rst:116]: @rmackay9
I will change it to be arbitrary then, with the note that for ROS-user they can place it in the catkin directory.

**Comment by thien94** [comment]
> thanks....okay, so when time comes, and I squash before merge, I will drop the Sphinx commit from this PR....

Hi @Hwurzburg , should I remove the Sphinx commit from this PR?

**Comment by Hwurzburg** [comment]
yes, or I will when I merge it....Pierre says hes got it covered

**Comment by Hwurzburg** [comment]
Merging...thanks...

**Pull request merged**